sql/core/src/test/resources/sql-tests/inputs/cte-nested.sql

```diff
@@ -103,3 +103,11 @@ SELECT (
     SELECT * FROM t
   )
 );
+
+-- CTE in subquery expression shadows outer 4
```

Comment:
This comment is correct about `subquery expression` but many comments in this file are not correct.

For example, the following query doesn't have subquery expression
```
-- CTE in subquery expression shadows outer
WITH t AS (SELECT 1)
SELECT (
  WITH t AS (SELECT 2)
  SELECT * FROM t
);
```